Add Spotify retrive URI instructions

starkillerOG opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

Add aditional instructions, after clearifing in the discord chat


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase:
- This PR fixes or closes issue:

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

starkillerOG committed:
Add Spotify retrive URI instructions

frenck commented on source/_integrations/spotify.markdown:
```diff
@@ -72,6 +72,7 @@ After you have set up the above:
 
 You can send playlists to Spotify using the `"media_content_type": "playlist"`, which are part of the
 [media_player.play_media](/integrations/media_player/#service-media_playerplay_media) service.
+The `media_content_id` can be obtained from the spotify desktop-app by clicking on the more options ("...") next to the album art picture, selecting "share" and then "Copy Spotify URI". This schould look like: "spotify:playlist:??????????????????????". The id indicated by the ??? can also be retrived from the last part of "share" --> "Copy Playlist Link" (also available in the spotify phone and w ...
```
This is incorrect. you can use any Spotify link.

starkillerOG replied:
I corrected it, is this better?

starkillerOG committed:
correction to retrieving media_content_id

examle playlist: "Love Songs & Chill"
"Copy Spotify URI" gives: spotify:playlist:4QuJ2DbcTe7R8lzqfNXz7v
this works as `media_content_id`
"Copy Playlist Link" gives: https://open.spotify.com/playlist/4QuJ2DbcTe7R8lzqfNXz7v?si=cUKngvAER3KK3BHgNayOEA
this does not work as `media_content_id` and gives an error in HomeAssistant
However if you remove everthing after the question mark ? is does work:
https://open.spotify.com/playlist/4QuJ2DbcTe7R8lzqfNXz7v

frenck commented on source/_integrations/spotify.markdown:
```diff
@@ -72,6 +72,7 @@ After you have set up the above:
 
 You can send playlists to Spotify using the `"media_content_type": "playlist"`, which are part of the
 [media_player.play_media](/integrations/media_player/#service-media_playerplay_media) service.
+The `media_content_id` can be obtained from the spotify desktop-app by clicking on the more options ("...") next to the album art picture, selecting "share" and then "Copy Spotify URI" or "Copy Playlist Link" (also available in the spotify phone and web app). If you use "Copy Playlist Link" remove everything after and including the question mark '?' in the link.
```
```suggestion
The `media_content_id` can be obtained from the Spotify desktop-app by clicking on the more options ("...") next to the album art picture, selecting "share" and then "Copy Spotify URI" or "Copy Playlist Link" (also available in the Spotify phone and web app). If you use "Copy Playlist Link" remove everything after and including the question mark '?' in the link.
```

- Interesting, so leaving in query parameters in the URI doesn't work? If so, that should be fixed in the integration to handle that automatically and nicely. I feel like we should avoid adding notices like that to the docs and make improve the experience instead.

starkillerOG replied:
@frenck Yes you are correct: leaving in query parameters in the URI doesn't work.
I agree that it would be nicer if the integration itself could just handle it and internally remove everything after the "?" itself. Schould not be to difficult I think.
Could just use a .split("?") or something like that.

starkillerOG committed:
Capital Spotify

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

starkillerOG committed:
Not needed anymore since this has been fixed

starkillerOG commented:
@frenck can this be merged, or schould I close this PR?

frenck commented:
@starkillerOG It's still on the list. Please have a bit of patience, thanks.

frenck reviewed (approved):
Thanks, @starkillerOG!

I've opened up https://github.com/home-assistant/core/pull/33084 to address the issue from the code base end.

This PR can be merged in the current state 👍
